src/EIP7702Proxy.sol

```diff
@@ -1,88 +1,90 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.0;
 
+/// @review does Solady v0.1 have something equivalent offerings for these depedencies? would be nice to use what we recently paid to audit and would probably also be optimized
 import {Proxy} from "openzeppelin-contracts/contracts/proxy/Proxy.sol";
 import {ERC1967Utils} from "openzeppelin-contracts/contracts/proxy/ERC1967/ERC1967Utils.sol";
 import {ECDSA} from "openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
 import {Address} from "openzeppelin-contracts/contracts/utils/Address.sol";
 
 /// @title EIP7702Proxy
-/// @notice Proxy contract designed for EIP-7702 smart accounts
-/// @dev Implements ERC-1967 with an initial implementation and guarded initialization
+/// @notice Proxy contract designed for EIP-7702 smart accounts.
+/// @dev Implements ERC-1967 with an initial implementation and guarded initialization.
+/// @author Coinbase (https://github.com/base-org/eip-7702-proxy).
 contract EIP7702Proxy is Proxy {
-    // ERC1271 interface constants
+    /// @notice ERC1271 interface constants
     bytes4 internal constant ERC1271_MAGIC_VALUE = 0x1626ba7e;
-    bytes4 internal constant ERC1271_ISVALIDSIGNATURE_SELECTOR = 0x1626ba7e;
     bytes4 internal constant ERC1271_FAIL_VALUE = 0xffffffff;
 
     /// @notice Address of this proxy contract (stored as immutable)
     address immutable proxy;
+
     /// @notice Initial implementation address set during construction
     address immutable initialImplementation;
+
     /// @notice Function selector on the implementation that is guarded from direct calls
     bytes4 immutable guardedInitializer;
 
+    /// @review natspec
     event Upgraded(address indexed implementation);
 
+    /// @review natspec
     error InvalidSignature();
+
+    /// @review natspec
     error InvalidInitializer();
+
+    /// @review natspec
     error InvalidImplementation();
 
+    /// @review natspec
     constructor(address implementation, bytes4 initializer) {
         proxy = address(this);
         initialImplementation = implementation;
+        /// @review what happens if `initializer` is one of the selectors we define in this contract?
```

Comment:
flagging

Reply:
this is a really interesting thought... I actually think that if the guarded initializer matches a function selector defined on the proxy, things would continue to work as expected because the fallback would never be hit and thus the guard would never be hit. when calling `initialize`, this is desired behavior. but if the guarded initializer were `"isValidSignature(bytes32, bytes)"`, then presumably the implementation could be initialized by calling this on the proxy. it's a very specific case that doesn't seem like it would arise by accident.... but maybe it's worth doing a check against the value of `initializer` in the constructor and to prevent that from matching any public function in the proxy apart from `initialize` itself? what do you think?